app/models/github_sync_service.rb

```diff
@@ -0,0 +1,191 @@
+class GithubSyncService
```

Comment:
This class is very well organized and you did a great job on adding documentation to make sure whoever is reading this can understand it 👍 

I'm interested to know how we can improve this architecture, because it seams a bit coupled since all the logic to sync with Github and the logic to process a webhook is here.

Imagine that our database is larger enough to the point where we need to do some sort of pagination when syncing with Github, we can't use this code as it is because for example most of the methods, like `update_authors_synced_with_issue `, make use of `github_issues`.

Could you try to decouple this class a bit please? I'm happy to discuss some ideas first if you want.

Reply:
This was discussed and agreed on slack